File: backend/app/schemas/event.py
"""RecruitmentEvent schemas."""

from __future__ import annotations

from datetime import date, time

from pydantic import BaseModel

from app.models.enums import EventStatus
from app.schemas.common import ORMModel


class EventBase(BaseModel):
    title: str
    description: str | None = None
    event_date: date
    start_time: time | None = None
    end_time: time | None = None
    location: str | None = None
    university_id: int | None = None
    event_type: str
    attendees_count: int = 0
    notes: str | None = None
    latitude: float | None = None
    longitude: float | None = None


class EventCreate(EventBase):
    status: EventStatus = EventStatus.SCHEDULED


class EventUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    event_date: date | None = None
    start_time: time | None = None
    end_time: time | None = None
    location: str | None = None
    university_id: int | None = None
    event_type: str | None = None
    status: EventStatus | None = None
    attendees_count: int | None = None
    notes: str | None = None
    latitude: float | None = None
    longitude: float | None = None


class EventOut(ORMModel):
    id: int
    title: str
    description: str | None = None
    event_date: date
    start_time: time | None = None
    end_time: time | None = None
    location: str | None = None
    university_id: int | None = None
    event_type: str
    status: str
    attendees_count: int
    notes: str | None = None
    latitude: float | None = None
    longitude: float | None = None

File: ios/Det695/Models/Event.swift
import Foundation

/// Status of a detachment outreach event. Decodes defensively so an unexpected
/// server value never crashes the list.
enum EventStatus: String, CaseIterable, Identifiable {
    case scheduled, completed, cancelled

    var id: String { rawValue }

    static func from(_ raw: String) -> EventStatus {
        EventStatus(rawValue: raw.lowercased()) ?? .scheduled
    }

    var label: String { rawValue.prefix(1).uppercased() + rawValue.dropFirst() }
}

/// A detachment outreach event. Mirrors the backend `EventOut` schema. Dates and
/// times arrive as plain strings (`event_date` = `YYYY-MM-DD`, `start_time` /
/// `end_time` = `HH:MM:SS`) and are kept as `String`.
struct EventOut: Decodable, Identifiable {
    let id: Int
    let title: String
    var description: String?
    let eventDate: String
    var startTime: String?
    var endTime: String?
    var location: String?
    var universityId: Int?
    let eventType: String
    let status: String
    let attendeesCount: Int
    var latitude: Double?
    var longitude: Double?
    var notes: String?

    var statusValue: EventStatus { .from(status) }
}
